### PR TITLE
[feature] Update Add-on User Settings UI

### PR DESCRIPTION
**Pull request by samanehsan**
Continuation of https://github.com/CenterForOpenScience/osf.io/pull/2414

### CODE REVIEW

**Comment by sloria** [comment]
Once this is up to date with develop, I will continue with the next pass of review.

**Comment by samanehsan** [comment]
@sloria, this is up to date and ready for another pass! :checkered_flag:

**Comment by sloria** [comment]
The logo images are much larger than they need to be. Can you resize and compress them?

**Comment by sloria** [comment]
There are FOUCs of the knockout-punches code happening:

addon_fouc

The unstyled text should be hidden, either by using `scripted` or using a regular `data-bind="text: ..."`.

**Comment by GaryKriebel** [comment]
I mentioned to @samanehsan earlier that Dataverse "Connect an account" is different than the others.
screen shot 2015-07-02 at 10 37 02 am

**Comment by GaryKriebel** [comment]
Additionally, there seems to be font differences between the name of each addon and the "Connect Account"....

See GithHub, Google Drive and Mendeley rows.

File: website/addons/twofactor/static/twoFactorUserConfig.js [hunk 1]
```diff
@@ -1,63 +1,193 @@
...
+            $(self.qrCodeSelector).qrcode(self.urls.otpauth);
+        }
+    }
+};
+
+ViewModel.prototype.fetchFromServer = function() {
+    var self = this;
+    return $.getJSON(self.settingsUrl)
+        .then(function(response) {
+            return response.result;
```
Comment (sloria) [inline website/addons/twofactor/static/twoFactorUserConfig.js]: @samchrisinger I'm not sure that this is doing anything; can it be removed?

Comment (samchrisinger) [inline website/addons/twofactor/static/twoFactorUserConfig.js]: This unpacks the data, and is needed.

File: website/addons/twofactor/static/twoFactorUserConfig.js [hunk 2]
```diff
@@ -1,63 +1,193 @@
...
-                self.messageClass('text-info');
-            }, timeout);
+
+    self.isEnabled = ko.observable(false);
+    self.isConfirmed = ko.observable(false);
+    self.secret = ko.observable('');
+
+    self.urls = {};
+}
+$.extend(ViewModel.prototype, ChangeMessageMixin.prototype);
```
Comment (sloria) [inline website/addons/twofactor/static/twoFactorUserConfig.js]: @samanehsan Doesn't the constructor of `ChangeMessageMixin` need to be called in the `ViewModel` constructor?

Comment (sloria) [inline website/addons/twofactor/static/twoFactorUserConfig.js]: That should initialize the `message` and `messageClass` variables, so you would be able to remove their initialization on ~L21-22.

**Comment by sloria** [comment]
The orange button doesn't seem right here:

osf___account_settings

We don't have any other orange buttons on the site. Also, the horizontal rules around the button are misleading (horizontal rules delineate sections). 

Why not use the same red links that are used for the addons? 

osf___configure_add-on_accounts

**Comment by sloria** [comment]
Pass finished.

**Comment by sloria** [comment]
@GaryKriebel Saman has already addressed the Dataverse language in https://github.com/samanehsan/osf.io/commit/c468dfe151e70d1688a3e7a2221dd9790281ce76. I'm not seeing the font issue; do you mean that the font for the addon names is different "Connect Account"? This is expected; we are using Bootstrap's defaults for headings and links.

**Comment by sloria** [comment]
That said, I'm not sure the add-on names need to be headings; they take up a lot of space, which may become an issue if/when we add more add-ons.

**Comment by GaryKriebel** [comment]
Actually, I meant the addon names with respect to themselves (which upon second look seem to be fine) and the "Connect Account" with respect to themselves.

The bottom two instances look bigger than the 3rd & 4th from the bottom.

screen shot 2015-07-02 at 1 49 32 pm

**Comment by sloria** [comment]
OK, thanks for the clarification @GaryKriebel . @samanehsan Can you take a look at the font sizes there? Also, the casing needs to be consistent; the "a" should be capitalized for all items.

**Comment by GaryKriebel** [comment]
Goodness... I didn't even notice the "a" .... it was a remnant from the original Dataverse wording change

**Comment by sloria** [comment]
Dataverse isn't currently working in this PR. The `/settings/dataverse/` endpoint is returning a 400 response. I think this is caused by the `must_have_addon` decorator. The view will need to be updated.

**Comment by sloria** [comment]
Once the above issues are addressed, I will do another pass.

**Comment by samanehsan** [comment]
@sloria, with the FOUC for the Mendeley and Zotero, when I use the `scripted` tag nothing loads and using `data-bind` doesn't solve the problem either. Maybe it has something to do with the fact that some of the observables are in deeply nested view models. Is there anything else I can try?

**Comment by samanehsan** [comment]
Also, how can I easily compress PNG files? I made the addon images smaller, but I have yet to compress them.

**Comment by samanehsan** [comment]
Besides that, I've addressed all other comments! :sunflower:

**Comment by sloria** [comment]
@samanehsan It's fine to have a minor FOUC as long as the KO punches syntax isn't showing. Using data-bind should address that problem.

ImageOptim is a simple program for compressing images.

File: website/templates/profile/user_settings_default.mako [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
+    </h4>
     <!-- ko foreach: accounts -->
-
-
     <table class="table">
         <thead>
             <tr>
-                <th>Authorized as <a href="{{ profileUrl }}"><em>{{ name }}</em></a></th>
-                <th><a data-bind="click: $root.askDisconnect" class="text-danger">Delete Access Token</a></th>
+                <th>Authorized by <a data-bind="attr.href: profileUrl, text: name"></a></th>
```
Comment (sloria) [inline website/templates/profile/user_settings_default.mako]: The name here used to be italicized, which I think is preferred. Can you add it back?

Comment (samanehsan) [inline website/templates/profile/user_settings_default.mako]: Ok, should I make all of them italic and link to either the OSF profile or the add-on profile? Here's what it looks like now: 
screen shot 2015-07-06 at 5 32 04 pm

**Comment by samanehsan** [comment]
This is ready for another pass @sloria! :watermelon:

File: website/templates/profile/user_settings_default.mako (hunk 3 above)
Comment (sloria) [inline website/templates/profile/user_settings_default.mako]: All external usernames should be italicized. Also, Mendeley should be made consistent with the rest of the addons.

**Comment by sloria** [comment]
Once, https://github.com/CenterForOpenScience/osf.io/pull/3409#discussion_r34042687 is addressed, this should be good to merge.

**Comment by samanehsan** [comment]
Just made that change @sloria. Here's what it looks like now: 
screen shot 2015-07-07 at 4 57 40 pm

**Comment by sloria** [comment]
@samanehsan Can you make the position and casing for Mendeley and Dataverse consistent with the other addons?

**Comment by samanehsan** [comment]
@sloria, I'm not sure how that would work since users can authorize multiple Dataverses and Mendeley accounts. E.g: 
screen shot 2015-07-07 at 5 05 56 pm

**Comment by sloria** [comment]
Ah ok, that's right. You can keep the casing as is.

**Comment by samanehsan** [comment]
Ok, in that case I think this should be good to go :sunny:

**Comment by sloria** [comment]
Discussed in brief with @GaryKriebel : One side effect of this PR is that it introduces some inconsistency with the project settings UI. 

As a follow-up to this PR, the project settings UI should also be updated. Can we plan/assign this, @lbanner @GaryKriebel ?

For now, just adding the new logo icons to the project settings would go a long way; @samanehsan would these be quick to add to the project settings?

**Comment by lbanner** [comment]
Yeah -- that followup is next on list. At this point, I'd think it could be @samanehsan or @samchrisinger, but @samchrisinger needs to get thru Pre-Reg prize.

**Comment by samanehsan** [comment]
@sloria, the icons should be a quick addition

**Comment by sloria** [comment]
@samanehsan Great. Once those are added an @lbanner and @GaryKriebel give the "go" after their functional review, this can be merged. :ear_of_rice:

**Comment by lbanner** [comment]
OK ... @samanehsan  -- Those will be added as first step in bring up-to-date Project Settings, correct?

**Comment by samanehsan** [comment]
I thought @sloria was suggesting that I add those in now to give some consistency.

**Comment by lbanner** [comment]
That's correct - just clarifying that we are talking Project Settings, Add-ons.

**Comment by samanehsan** [comment]
@sloria, done adding in those icons. I also added one for external links: 
screen shot 2015-07-09 at 10 08 12 am

**Comment by sloria** [comment]
Thanks. We use that icon for project links; the "external link" icon might be more appropriate.

**Comment by samanehsan** [comment]
What's the "external link" icon?

**Comment by sloria** [comment]
https://fortawesome.github.io/Font-Awesome/icon/external-link/

**Comment by samanehsan** [comment]
Oh, okay! I just made that change @sloria: 
screen shot 2015-07-09 at 10 15 52 am

**Comment by sloria** [comment]
Thanks! 

Merged into release/0.43.0.

**Pull request closed without merging**